src/Microsoft.AspNet.Mvc.Formatters.Json/Logging/JsonInputFormatterLoggerExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Mvc.Formatters.Json.Logging
+{
+    internal static class JsonInputFormatterLoggerExtensions
+    {
+        private static readonly Action<ILogger, string, Exception> _jsonInputFormatterCrashed;
+
+        static JsonInputFormatterLoggerExtensions()
+        {
+            _jsonInputFormatterCrashed = LoggerMessage.Define<string>(
+                LogLevel.Verbose,
+                1,
+                "Json input formatting failed, exception was '{0}'.");
```

Comment:
Ah, good point on the exception. I figured the message might not pass muster but couldn't come up with a better one mostly because I was trying to shoe-horn the exception in there.